core/services/directrequest/delegate.go

```diff
@@ -126,9 +125,9 @@ func (l *listener) Start() error {
 	return l.StartOnce("DirectRequestListener", func() error {
 		unsubscribeLogs := l.logBroadcaster.Register(l, log.ListenerOpts{
 			Contract: l.oracle,
-			Logs: []generated.AbigenLog{
-				oracle_wrapper.OracleOracleRequest{},
-				oracle_wrapper.OracleCancelOracleRequest{},
+			LogsWithTopics: map[common.Hash][][]log.Topic{
```

Comment:
IIRC, `indexed` topics are hashed. Should definitely test this out on a testnet

Reply:
@spooktheducks so if i understand this right, only parameters with `indexed` keyword are stored in the Topics array (others would be in Data).

And then:
"Indexed event parameters that are not value types, i.e. arrays and structs are not stored directly but instead a keccak256-hash of an encoding is stored"

So to filter for an array or a struct, a hash of it should be used in the filter instead.
https://docs.soliditylang.org/en/latest/abi-spec.html#abi-events
https://docs.soliditylang.org/en/latest/abi-spec.html#indexed-event-encoding

Reply:
i'll mention it in the doc